### PR TITLE
Scancode enums

### PR DESCRIPTION
**Pull request by iggyglass**
I added two enums (and some functions to go along with them) containing PS2 scancodes.

One of the enums contains the "non-special" characters (escape, tab, alphabet, etc.), while the other contains the special characters prefixed by `0xE0`.

### CODE REVIEW

**Comment by Harvie** [comment]
Hello,
thank you for your contribution. i will review the code...

File: src/ps2dev.cpp [hunk 1]
```diff
@@ -9,307 +9,401 @@
...
+ * These functions set the state of internal pull-up resistors
  */
-void
-PS2dev::gohi(int pin)
+void PS2dev::gohi(int pin)
 {
-  pinMode(pin, INPUT);
-  digitalWrite(pin, HIGH);
+	DDRB &= ~(1 << pin); // pinMode(pin, INPUT);
+	PORTB |= (1 << pin); // digitalWrite(pin, HIGH);
```
Comment (Harvie) [inline src/ps2dev.cpp:49]: This is AVR specific code, which will break compatibility with other microcontrollers that i aim to support. there is reason for arduino HAL. If you absolutely need this, we can add our own HAL with set of defines in separate header file which will be chosen at compile-time based on #ifdef ARDUINO or something...

File: src/ps2dev.h [hunk 2]
```diff
@@ -30,5 +177,4 @@ class PS2dev
 		void ack();
 };
 
-#endif /* ps2dev_h */
-
+#endif /* ps2dev_h */
```
Comment (Harvie) [inline src/ps2dev.h:180]: gentlemen leaves the newline at the end of the file :-)

File: src/ps2dev.h [hunk 3]
```diff
@@ -1,27 +1,174 @@
...
+ * modified by Tomas 'Harvie' Mudrunka 2019.
+ * modified once more by Iggy Glassman 2020.
  * Release into public domain.
  */
 
 #ifndef ps2dev_h
 #define ps2dev_h
 
 #include "Arduino.h"
+#include <avr/io.h>
```
Comment (Harvie) [inline src/ps2dev.h:15]: same thing. does not look portable to me.

File: src/ps2dev.cpp [hunk 4]
```diff
@@ -9,307 +9,401 @@
...
+  		case 0xF0: //set scan code set
+    		ack();
+    		if (!read(&val)) ack(); //do nothing with the rate
+    		break;
+  		case 0xEE: //echo
+    		write(0xEE);
+    		break;
+  		case 0xED: //set/reset LEDs
+    		ack();
+    		if (!read(leds)) ack(); //do nothing with the rate
```
Comment (Harvie) [inline src/ps2dev.cpp:269]: you reformating my code makes it bit hard to actualy see what changes you made vs what is just reformating. i think we can agree on using some clang-format config, but changing formating SHOULD be done as separate pull request, so it's obvious that it only changes the style. but i would prefer you to not change the style, or at least provide the .clang-format file.

Comment (iggyglass) [reply src/ps2dev.cpp:269]: Ok!

**Comment by iggyglass** [comment]
I will close this pull request and make the changes requested!

(Apologies for the reformatting thing -- this is my first pr.)

File: src/ps2dev.cpp (hunk 1 above)
Comment (iggyglass) [reply src/ps2dev.cpp:49]: Sounds good.

File: src/ps2dev.cpp (hunk 4 above)
Comment (iggyglass) [reply src/ps2dev.cpp:269]: That makes sense.

**Comment by Harvie** [comment]
> I will close this pull request and make the changes requested!

Thanks. BTW if you have unmerged pullrequest and you make some changes to your fork, i believe the pull request gets updated automaticaly. So no need to recreate the PR. But it does not really matter, just for your convenience...
